components/example-notebook-servers/README.md

```diff
@@ -0,0 +1,139 @@
+dependencies and common utilities from APT, installs `kubectl`, creates the
+user account for the downstream images and adds the
+[S6-overlay](https://github.com/just-containers/s6-overlay) for the init system.
+From this base image, images for Jupyter, RStudio and Visual Studio Code
+(through [code-server]([https://github.com/cdr/code-server)) are created.
+From the Jupyter image, images are created that show users how to install
+TensorFlow and PyTorch both with and without support for NVIDIA GPUs.
+In turn, these images are extended with an opinionated set of common tools
+and packages to help new users to quickly get started with Kubeflow.
+
+**Lineage of the provided example Notebook Server Images:**
+
+![Flow Chart of example Notebook Servers](./kubeflow-example-notebook-servers.png)
+
+## Creating custom Notebook Server images
+
+Because the needs of users varies greatly, the provided images were created to
+be easily extended by using the [S6-overlay](https://github.com/just-containers/s6-overlay)
+init system and including [Conda](https://github.com/conda-forge/miniforge).
+
+### Installing packages
+
+The most common customization to Notebook Server images will be to install
+a selection of packages specific that will be needed in the Notebook Server.
+Users can install packages in a running Notebook Server using `pip install ...`
+or `conda install ...`. However, packages installed in this manner will not
+persist if the Notebook Server is stopped or the Pod is restarted.
+While it is possible to install Python packages in the user's home directory
+by using `pip install --user ...`, this can cause unexpected behaviour if the
+workspace volume is later used in a different Notebook Server, as the packages
+present in the home directory might be used rather than the ones present in the
+Notebook Server's image. For this reason it is recommended to create custom
+images if persistance of the installed packages is wanted. An example of such
+a Dockerfile can be seen (POINT TO A DOCKERFILE THAT USES PIP AND ONE THAT USES CONDA).
+If APT dependencies are required need to be installed, this must be done with by
+creating a custom Notebook Server image as the user in the provided images does
+not have `sudo` privileges. To do so, create a Dockerfile and choose a given
+base image. Next, change to the root user using `USER root` and make any
+changes that require root privileges. Once this is done, switch back to the
+regular user with `USER $NB_UID` and, if needed, install the wanted packages
+using Pip or Conda.
+
+### Custom startup scripts and extra services (Advanced)
+
+Some users might require custom scripts to run during the startup of the
+Notebook Server container, or advanced users might want to add additional
+services that run inside the container (for example: an Apache or NGINX web server).
+To make this easy, the [S6-overlay](https://github.com/just-containers/s6-overlay)
+is used in the provided images.
```

Comment:
Another example of passive to active: `To make this easy, we use the S6-overlay as init-system`